stable/yugabyte/templates/certificates.yaml

```diff
@@ -0,0 +1,123 @@
+{{- $root := . -}}
+---
+{{- if $root.Values.tls.certManager.enabled }}
+{{- if not $root.Values.tls.certManager.useClusterIssuer }}
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name:  {{ (printf "%s-bootstrap" (include "yugabyte.tls_cm_issuer" $root)) | quote }}
+  namespace: "{{ $root.Release.Namespace }}"
+spec:
+  selfSigned: {}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: {{ $root.Values.oldNamingStyle | ternary "yugabyte-ca" (printf "%s-ca" (include "yugabyte.fullname" $root)) }}
+  namespace: "{{ $root.Release.Namespace }}"
+spec:
+  isCA: true
+  privateKey:
+    algorithm: {{ $root.Values.tls.certManager.certificates.algorithm | quote }}
+    encoding: PKCS8
+    size: {{ $root.Values.tls.certManager.certificates.keySize }}
+  commonName: Yugabyte Selfsigned CA
+  secretName: {{ $root.Values.oldNamingStyle | ternary "yugabyte-ca" (printf "%s-ca" (include "yugabyte.fullname" $root)) }}
+  issuerRef:
+    name: {{ (printf "%s-bootstrap" (include "yugabyte.tls_cm_issuer" $root)) | quote }}
+    kind: Issuer
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: {{ include "yugabyte.tls_cm_issuer" $root | quote }}
+  namespace: "{{ $root.Release.Namespace }}"
+spec:
+  ca:
+    secretName: {{ $root.Values.oldNamingStyle | ternary "yugabyte-ca" (printf "%s-ca" (include "yugabyte.fullname" $root)) }}
+---
+{{- end }}
+
+
+{{- range .Values.Services }}
+{{- $service := . -}}
+{{- $appLabelArgs := dict "label" .label "root" $root -}}
+{{- $serviceValues := (dict "Service" $service "Values" $root.Values "Chart" $root.Chart "Release" $root.Release) -}}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: {{ $root.Values.oldNamingStyle | ternary (printf "%s-yugabyte-tls-cert" $service.label) (printf "%s-%s-tls-cert" (include "yugabyte.fullname" $root) $service.label) }}
+  namespace: "{{ $root.Release.Namespace }}"
+spec:
+  secretTemplate:
+    labels:
+      {{- include "yugabyte.applabel" ($appLabelArgs) | indent 6 }}
+      {{- include "yugabyte.labels" $root | indent 6 }}
+  issuerRef:
+    name:  {{ include "yugabyte.tls_cm_issuer" $root | quote }}
+    {{- if $root.Values.tls.certManager.useClusterIssuer }}
+    kind: ClusterIssuer
+    {{- else }}
+    kind: Issuer
+    {{- end }}
+  secretName: {{ $root.Values.oldNamingStyle | ternary (printf "%s-yugabyte-tls-cert" $service.label) (printf "%s-%s-tls-cert" (include "yugabyte.fullname" $root) $service.label) }}
+  duration: {{ $root.Values.tls.certManager.certificates.duration | quote }}
+  renewBefore: {{ $root.Values.tls.certManager.certificates.renewBefore | quote }}
```

Comment:
It seems like cert manager will update this cert at renewal time, which presumbly gets reflected on files within the pod. Right now, there is no automatic reload of the certs, so certs are not actually reloaded until a restart happens. Even when https://github.com/yugabyte/yugabyte-db/issues/9443 is implemented, an explicit signal or RPC will be required to achieve this reload. To do so automatically would probably require some kind of a 'hook'

I would add this information as a comment in values.yaml to make it clear that renewal period can be set but has to followed up with an explicit restart.

Reply:
This is indeed an issue, which we could reproduce with certificates which had a 2h lifetime.

I will add a comment to the `values.yaml`